frontend/src/components/advanced-auto-filter/brand-details/brand-details.tsx

```diff
@@ -18,42 +20,43 @@ import styles from './styles.module.scss';
 
 type Props = {
   id: string;
-  selectedBrandId: string;
-  selectedModelId: string;
   onBrandDetailsChange: (data: BrandDetailsType) => void;
   onBrandDetailsRemove?: () => void;
 };
 
 const BrandDetails: FC<Props> = ({
   id,
-  selectedBrandId,
-  selectedModelId,
   onBrandDetailsChange,
   onBrandDetailsRemove,
 }) => {
+  const { brandDetails } = useAppSelector((state) => state.carFilter);
+  const { brandId, modelIds } = brandDetails.find(
+    (item) => item.id === id,
+  ) as BrandDetailsType;
```

Comment:
`.find` is a pretty heavy JS operation. Try to not use it outside of the `useMemo` or the functions in the React components